web/clusters.go

```diff
@@ -11,6 +12,182 @@ import (
 	"github.com/trento-project/trento/internal/hosts"
 )
 
+type Node struct {
+	Name       string
+	Attributes map[string]string
+	Resources  []*Resource
+	Ip         string
+	Health     string
+	VirtualIps []string
+}
+
+type Resource struct {
+	Id        string
+	Type      string
+	Role      string
+	Status    string
+	FailCount int
+}
+
+type Nodes []*Node
+
+func stoppedResources(c *cluster.Cluster) []*Resource {
+	var stoppedResources []*Resource
+
+	for _, r := range c.Crmmon.Resources {
+		if r.NodesRunningOn == 0 && !r.Active {
+			resource := &Resource{
+				Id: r.Id,
+			}
+			stoppedResources = append(stoppedResources, resource)
+		}
+	}
+
+	return stoppedResources
+}
+
+func (node *Node) Role() string {
+	if r, ok := node.Attributes["hana_prd_roles"]; ok {
+		role := r[strings.LastIndex(r, ":")+1:]
+		return strings.Title(role)
+	}
+	return "-"
+}
+
+func (node *Node) HealthState() string {
```

Comment:
What do you parse here?
Assuming something like `1:P:master1::worker:`, the first value? What does this number mean?

Reply:
The first number is the `SAPHanaSR health state`



4 | SAP HANA database is up and OK. The cluster does interpret this as a correctly running database.
-- | --
3 | SAP HANA database is up and in status info. The cluster does interpret this as a correctly running database.
2 | SAP HANA database is up and in status warning. The cluster does interpret this as a correctly running database.
1 | SAP HANA database is down. If the database should be up and is not down by intention, this could trigger a takeover.
0 | Internal Script Error – to be ignored.